src/server/debug/server/util.go

```diff
@@ -95,9 +92,7 @@ func collectDebugStream(tw *tar.Writer, r io.Reader, prefix ...string) (retErr e
 		return errors.EnsureStack(err)
 	}
 	defer func() {
-		if err := gr.Close(); retErr == nil {
-			retErr = err
-		}
+		multierr.AppendInto(&retErr, gr.Close())
```

Comment:
Curious why in most places we first initialize an error with `var errs error` before calling  `multierr.AppendInto(&errs, ...)`, but here we reuse the `retErrr`?

Reply:
There are two cases. 1 is where we are creating the raw return error from the sub-task, that's the `var errs error` case.  We just need to allocate a pointer to an interface to allocate the multi-error into.  2 is where we do some cleanup while returning from a function, in which case `func X() (retErr error)` allocates `retErr` and it's either the main error or nil; we append into that object.  (It may seem weird to write into a nil pointer, but the interface is not nil; it's an error that points to a nil object, as opposed to `var error *error` which is a nil pointer to an interface, not an interface pointing to a nil object.)